Widgets/Bars/ClassBar.lua

```diff
@@ -83,6 +84,18 @@ function W.UpdateClassBarWidget(button, unit, setting, subSetting, ...)
     if not setting or setting == const.OPTION_KIND.SIZE then
         widget:SetSizeStyle(styleTable.size)
     end
+    if not setting or setting == const.OPTION_KIND.HIDE_OUT_OF_COMBAT then
+        widget.hideOutOfCombat = styleTable.hideOutOfCombat
+        widget:UpdateEventListeners()
+        local shouldShow = widget:ShouldShow()
+        if shouldShow then
+            widget:ShowBars()
+            widget:TogglePowerEvents(true)
+        else
+            widget:HideBars()
+            widget:TogglePowerEvents(false)
+        end
+    end
```

Comment:
This should just call `Update` after updating listeners.

Reply:
Done. I'm still a novice at Lua, so it took me a while to figure out that I should call update like this
`widget.Update(button)` and not like this `widget:Update()`  😅

Reply:
Yea makes sense, I'm not entirely satisfied with how it turned out, so should refactor it at some point..

Reply:
Sure, go ahead :)